File: AtC/ABC/075/D.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long

int main() {
	ll n,k; cin>>n>>k;
	vector<ll> x(n),y(n);
	for (ll i=0; i<n; i++)
		cin>>x[i]>>y[i];

	ll ans=~(1LL<<63);
	for (ll a=0; a<n; a++) {
		for (ll b=a; b<n; b++) {
			for (ll c=b; c<n; c++) {
				for (ll d=c; d<n; d++) {
					ll right = max({x[a],x[b],x[c],x[d]});
					ll left = min({x[a],x[b],x[c],x[d]});
					ll top = max({y[a],y[b],y[c],y[d]});
					ll bot = min({y[a],y[b],y[c],y[d]});
					ll area = abs((top-bot) * (right-left));
					ll contains=0;
					for (ll m=0; m<n; m++)
						if (x[m]>=left and x[m]<=right and y[m]>=bot and y[m]<=top)
							contains++;
					if (contains>=k and area<ans)
						ans=area;
				}
			}
		}
	}
	cout << ans << endl;
	return 0;
}

File: AtC/ABC/075/D.py
n,k = [int(x) for x in input().split()]
x = []
y = []
for i in range(n):
    xi,yi = [int(x) for x in input().split()]
    x.append(xi)
    y.append(yi)

ans = (2e9)**2

for i in range(n):
    for j in range(i+1,n):
        area = abs((x[i]-x[j]) * (y[i]-y[j]))
        contains = 0
        for k in range(n):
            if x[k] >= min(x[i],x[j])\
                    and x[k] <= max(x[i],x[j])\
                    and y[k] >= min(y[i],y[j])\
                    and y[k] <= max(y[i],y[j]):
                contains+=1
        if (contains>=k and area<ans):
            ans = area
print(ans)

